### PR TITLE
Ensures that the error content type matches the media type of the ret…

### PR DESCRIPTION
**Pull request by dbernstein**
…urrned representation.

Resolves https://jira.duraspace.org/browse/FCREPO-1888

### CODE REVIEW

File: fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/AbstractResourceIT.java [hunk 1]
```diff
@@ -268,9 +270,14 @@ protected static String getLocation(final HttpResponse response) {
     }
 
     protected String getContentType(final HttpUriRequest method) throws ClientProtocolException, IOException {
+        return getContentType(method, OK);
+    }
+
+    protected String getContentType(final HttpUriRequest method, final Status httpStatus) throws ClientProtocolException,
```
Comment (awoods) [inline fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/AbstractResourceIT.java]: Minor checkstyle error: https://travis-ci.org/fcrepo4/fcrepo4/builds/193642642#L6783

Comment (awoods) [inline fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/AbstractResourceIT.java]: This exception can be removed from the `throws` clause: 'ClientProtocolException'
..and its `import` declaration.

File: fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/AbstractResourceIT.java [hunk 2]
```diff
@@ -268,9 +270,14 @@ protected static String getLocation(final HttpResponse response) {
     }
 
     protected String getContentType(final HttpUriRequest method) throws ClientProtocolException, IOException {
```
Comment (awoods) [inline fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/AbstractResourceIT.java]: This exception can be removed from the `throws` clause: 'ClientProtocolException'

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/BadRequestExceptionMapper.java [hunk 3]
```diff
@@ -44,7 +45,7 @@
     public Response toResponse(final BadRequestException e) {
         LOGGER.error("BadRequestExceptionMapper caught an exception: {}", e.getMessage());
         debugException(this, e, LOGGER);
-        return status(BAD_REQUEST).entity(e.getMessage()).build();
+        return status(BAD_REQUEST).entity(e.getMessage()).type(DEFAULT_CONTENT_TYPE).build();
```
Comment (awoods) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/BadRequestExceptionMapper.java]: I think the clarity of the code would be increased by replacing `DEFAULT_CONTENT_TYPE` with the actual "TEXT_PLAIN_WITH_CHARSET".
I realize that by abstracting the actual content-type with a global variable allows for changing the value of that variable in the future... but in that theoretical scenario, there is an equally good chance that these classes will change as well and my sense would be the same then, too; namely, that seeing the actual content-type is more clear.

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/ExceptionMapperConstants.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+package org.fcrepo.http.commons.exceptionhandlers;
+
+import static org.fcrepo.http.commons.domain.RDFMediaType.TEXT_PLAIN_WITH_CHARSET;
+
+/**
+ * @author Daniel Bernstein
+ * @since Jan 19, 2017
+ */
+class ExceptionMapperConstants {
```
Comment (awoods) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/ExceptionMapperConstants.java]: Suggest removing this class.

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/BadRequestExceptionMapper.java (hunk 3 above)
Comment (dbernstein) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/BadRequestExceptionMapper.java]: Fair enough.

File: fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/ExceptionMapperConstants.java (hunk 4 above)
Comment (dbernstein) [inline fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/ExceptionMapperConstants.java]: It's no longer necessary - agreed.

**Pull request merged**